runtime/src/debug/cpp/KDebug.cpp

```diff
@@ -188,6 +207,119 @@ RUNTIME_USED const char* Konan_DebugGetTypeName(KRef obj) {
+
+  if (extendedTypeInfo == nullptr)
+    return nullptr;
+  if (operation <= 0 || operation >= extendedTypeInfo->debugOperationsCount_)
+    return nullptr;
+  return extendedTypeInfo->debugOperations_[operation];
+}
+
+}  // namespace
+
+extern "C" {
+
+RUNTIME_USED RUNTIME_WEAK char* Konan_DebugBuffer() {
+  return Konan_DebugBufferImpl();
+}
+
+RUNTIME_USED RUNTIME_WEAK int32_t Konan_DebugBufferSize() {
+  return Konan_DebugBufferSizeImpl();
+}
+
+RUNTIME_USED RUNTIME_WEAK char* Konan_DebugBufferWithObject(KRef obj) {
+  auto* impl = getImpl<char* (*)(KRef)>(obj, DO_DebugBufferWithObject);
+  if (impl == nullptr) return nullptr;
+  return impl(obj);
+}
+
+RUNTIME_USED RUNTIME_WEAK int32_t Konan_DebugBufferSizeWithObject(KRef obj) {
+  auto* impl = getImpl<int32_t (*)(KRef)>(obj, DO_DebugBufferSizeWithObject);
+  if (impl == nullptr) return 0;
+  return impl(obj);
+}
+
+// Auxilary function which can be called by developer/debugger to inspect an object.
+RUNTIME_USED RUNTIME_WEAK int32_t Konan_DebugObjectToUtf8Array(KRef obj, char* buffer, int32_t bufferSize) {
+  auto* impl = getImpl<int32_t (*)(KRef, char*, int32_t)>(obj, DO_DebugObjectToUtf8Array);
+  if (impl == nullptr) return 0;
+  return impl(obj, buffer, bufferSize);
+}
+
+RUNTIME_USED RUNTIME_WEAK int32_t Konan_DebugPrint(KRef obj) {
+  auto* impl = getImpl<int32_t (*)(KRef)>(obj, DO_DebugPrint);
+  if (impl == nullptr) return 0;
+  return impl(obj);
+}
+
+RUNTIME_USED RUNTIME_WEAK int32_t Konan_DebugIsArray(KRef obj) {
+  auto* impl = getImpl<int32_t (*)(KRef)>(obj, DO_DebugIsArray);
+  if (impl == nullptr) return 0;
+  return impl(obj);
+}
+
+RUNTIME_USED RUNTIME_WEAK int32_t Konan_DebugGetFieldCount(KRef obj) {
+  auto* impl = getImpl<int32_t (*)(KRef)>(obj, DO_DebugGetFieldCount);
+  if (impl == nullptr) return 0;
+  return impl(obj);
+}
+
+RUNTIME_USED RUNTIME_WEAK int32_t Konan_DebugGetFieldType(KRef obj, int32_t index) {
+  auto* impl = getImpl<int32_t (*)(KRef, int32_t)>(obj, DO_DebugGetFieldType);
+  if (impl == nullptr) return 0;
+  return impl(obj, index);
+}
+
+RUNTIME_USED RUNTIME_WEAK void* Konan_DebugGetFieldAddress(KRef obj, int32_t index) {
+  auto* impl = getImpl<void* (*)(KRef, int32_t)>(obj, DO_DebugGetFieldAddress);
+  if (impl == nullptr) return nullptr;
+  return impl(obj, index);
+}
+
+// Compute address of field or an array element at the index, or null, if incorrect.
+RUNTIME_USED RUNTIME_WEAK const char* Konan_DebugGetFieldName(KRef obj, int32_t index) {
+  auto* impl = getImpl<const char* (*)(KRef, int32_t)>(obj, DO_DebugGetFieldName);
+  if (impl == nullptr) return nullptr;
+  return impl(obj, index);
+}
+
+RUNTIME_USED RUNTIME_WEAK const char* Konan_DebugGetTypeName(KRef obj) {
+  auto* impl = getImpl<const char* (*)(KRef)>(obj, DO_DebugGetTypeName);
+  if (impl == nullptr) return nullptr;
+  return impl(obj);
+}
+
+RUNTIME_USED RUNTIME_WEAK
+void* Konan_DebugGetOperation(KRef obj, /* Konan_DebugOperation */ int32_t operation) {
```

Comment:
Is this public debug API actually required?

Reply:
Mostly for the completness of the debug API, but indeed, we may not expose it and enum.